atd-etl/afd_ems_import/afd_incident_upload.py

```diff
@@ -0,0 +1,384 @@
+#!/usr/bin/env python
+
+"""
+Name: AFD Incident Uploads
+Description: This flow uploads AFD Incident Response CSVs
+    The data is emailed to atd-afd-incident-data@austinmobility.io daily ~ 6:30AM. From there it
+    gets forwarded to a S3 bucket via AWS Simple Email Service.
+"""
+
+import datetime
+import json
+import boto3
+import pprint
+import os
+import time
+import email
+import tempfile
+import shutil
+import pandas
+from datetime import datetime, timedelta
+import psycopg2
+import psycopg2.extras
+from sshtunnel import SSHTunnelForwarder
+
+import onepasswordconnectsdk
+from onepasswordconnectsdk.client import Client, new_client
+
+pp = pprint.PrettyPrinter(indent=4)
+
+DEPLOYMENT_ENVIRONMENT = os.environ.get(
+    "ENVIRONMENT", "development"
+)  # our current environment from ['production', 'development']
+ONEPASSWORD_CONNECT_TOKEN = os.getenv("OP_API_TOKEN")  # our secret to get secrets 🤐
+ONEPASSWORD_CONNECT_HOST = os.getenv("OP_CONNECT")  # where we get our secrets
+VAULT_ID = os.getenv("OP_VAULT_ID")
+
+
+def main():
+    secrets = get_secrets()
+
+    global DB_USERNAME
```

Comment:
I know that I [left you hanging](https://github.com/cityofaustin/atd-vz-data/pull/1235/files#r1245923996) that last time I brought this up, but I'm starting to think that these global variables tie into the 1PW in the DAG vs 1PW in the container discussion. I'm starting to feel like the Docker containers should not be concerned with how the secrets are obtained but only with receiving them. It would help avoid a refactor if we ever shifted away from the 1PW server (which is unlikely and I surely hope we never have to!).

Like you mentioned today, it is a lot to think about, and I'll keep thinking about it over the weekend.

Reply:
in either case, i think you could slash all of these `global` declarations, `get_secrets()` outside of `main`, and declare these variables once in the global scope.

Reply:
John, thank you for your suggestion!

I had done this ugly thing with the `global` declarations inside `main()` because I was telling myself that it's good to get just as much out of the global execution as possible because that prevents it from being executed if someone were to `import` the script as a library. 

As I write this, I also think that the way it looks so 🤮 that it can't be right. I am going to ask about it today in dev sync, but I am increasingly feeling that I'm doing a lot of jumping-through-hoops for an esoteric reason that isn't really actually possible.

Reply:
In response to your question in dev sync, the thing that stood out to me was that it looks like variables are being declared and defined in one function and then being pushed up into global scope so that another function can read the values. I could be misunderstanding what is going on, but I would expect globally scoped variables to be declared in the global scope if multiple functions in that scope are reading them.

Reply:
i think that's a great point about not polluting the global scope for code that gets imported elsewhere. for scripts like this, i kind of prefer to just cram everything global up front at the top level so it's clear what all the dependencies are. 

either way (declaring globals in `main()` or in the global scope), maybe the real anti-pattern is not using `os.getenv()` and `os.setenv()` to pass around environment settings? 

i do not feel strongly about this at all, and think what you currently have is free of any red flags that would make maintaining the code difficult!

Reply:
Thank you both a ton for these thoughts! 🙏🏻 I really like how Mike put it about expecting global variables to be declared in the global scope -- he's identifying why just seeing `global` in the code looks strange. 

I cleaned up both scripts to make this look more normal. Because the `get_secets()` is still in a function, that needs to get declared first, but that's totally fine.